addons/auth_oauth/controllers/main.py

```diff
@@ -65,6 +65,7 @@ def list_providers(self):
                 redirect_uri=return_url,
                 scope=provider['scope'],
                 state=json.dumps(state),
+                nonce=base64.urlsafe_b64encode(os.urandom(16)),
```

Comment:
have you found cases where a missing `nonce` was blocking the flow? (out of curiosity)

Reply:
Yeah sadly I apparently noted this absolutely nowhere, but I think it was okta who failed if there was no nonce? One of okta and auth0 definitely did.

It annoys me a lot that I lost this bit.

Reply:
Confirmed, if I try to send a request to okta without `nonce` I get:
```python
{'state': ['{"d": "oauth", "p": 4, "r": "https%3A%2F%2F8c5b-212-68-194-130.ngrok.io%2Fweb"}'], 'error': ['invalid_request'], 'error_description': ["The 'nonce' parameter is required for authorize requests with either the 'id_token' or 'token' response types"]}
```